internal/agent/device/applications/controller.go

```diff
@@ -274,7 +275,7 @@ func (c *Controller) ensureEmbedded() error {
 		suffixPatterns := []string{"*.yml", "*.yaml"}
 		for _, pattern := range suffixPatterns {
 			// search for compose files
-			files, err := filepath.Glob(filepath.Join(lifecycle.EmbeddedComposeAppPath, element.Name(), pattern))
+			files, err := filepath.Glob(c.readWriter.PathFor(filepath.Join(lifecycle.EmbeddedComposeAppPath, element.Name(), pattern)))
```

Comment:
note: this was added to provide fileio support for testing and is not a logical change.